### PR TITLE
Fix big constant arrays not recognized as oversized

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/8775

### CODE REVIEW

File: tests/PHPStan/Analyser/data/bug-8775.php [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+ */
+class Test
+{
+	/**
+	 * @var Fixtures
+	 */
+	public static $fixtures;
+
+	function doFoo():void {
+		assertType('(int|string)', Test::$fixtures['257']);
```
Comment (bendavies) [inline tests/PHPStan/Analyser/data/bug-8775.php:276]: this appears to be the type before this fix? so the fix isn't actually tested here?
https://phpstan.org/r/2e0b1bf7-c51c-42ff-b6f4-eba9e384506c

Comment (bendavies) [reply tests/PHPStan/Analyser/data/bug-8775.php:276]: so is there no way for this to be marked as just an `int` then? 256 is a hard limit for type inference?

Comment (staabm) [reply tests/PHPStan/Analyser/data/bug-8775.php:276]: Before the fix the type was `int|string`. Now it is `(int|string)` which means its benevolent (more forgiving). 

It should prevent the followup errors. I cannot say for sure as the repro given in the origin issue does not show these followup errors. 

For arrays bigger then 256 elements we cannot be that precise anymore for performance reasons

Comment (bendavies) [reply tests/PHPStan/Analyser/data/bug-8775.php:276]: thanks, i didn't know what benevolent union was!

**Comment by ondrejmirtes** [comment]
Changed it a bit. Thank you.

**Pull request merged**